### PR TITLE
Further-regression-applied

### PR DESCRIPTION
**Pull request by BlayneJLees**
# Further-regression-applied #

## Description ##

Small change so all lines are now sorted and that regression is also applied to the snapped lines.
## Checklist ##

- [x] PEP8 compliant (run `black --check $(git diff --name-only HEAD origin/develop | grep .py) -l 119`);
- [x] Passing tests;
- [ ] _(If required)_ Confluence pages updated.

### CODE REVIEW

**Comment by BlayneJLees** [comment]
> Noticed that Jenkins is not configured for this package. Please sort this out.

Have added the file but im unfamiliar with the process, I take it this pr needs to be merged in order for the jenkins file to take effect?

**Comment by VirginieBfd** [comment]
> > Noticed that Jenkins is not configured for this package. Please sort this out.
> 
> Have added the file but im unfamiliar with the process, I take it this pr needs to be merged in order for the jenkins file to take effect?

1

**Comment by VirginieBfd** [comment]
> > Noticed that Jenkins is not configured for this package. Please sort this out.
> 
> Have added the file but im unfamiliar with the process, I take it this pr needs to be merged in order for the jenkins file to take effect?

@vgcornwell can you help @BlayneJLees setting up Jenkins as you have done recently with Vik for patch-detection? Cheers :)

**Comment by vgcornwell** [comment]
@VirginieBfd - @BlayneJLees and I will look into this this afternoon!

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,11 +13,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 
 ### Added
+- regression on smoothed lines
```
Comment (LTMeyer) [inline CHANGELOG.md:16]: Some of the listed points have been released already.

File: hummingbirdtech/planting_lines/line_reconstruction/algorithm/line_creator.py [hunk 2]
```diff
@@ -108,6 +108,7 @@ def get_lines(self, row_band, vegetation_band):
         self._logger.debug("smoothing the lines")
 
         smoothed_lines = self.smooth_lines(snapped_lines=snapped_lines, labels=labels)
+        smoothed_lines = [LineString(sorted(line.coords)) for line in smoothed_lines]
```
Comment (LTMeyer) [inline hummingbirdtech/planting_lines/line_reconstruction/algorithm/line_creator.py:111]: This should be added to the `smooth_lines` function called above. The reason being here we gonna loop twice through the set, while the same operation could be done in only one go.

**Comment by VirginieBfd** [comment]
@BlayneJLees, tests are failling because there is no test :laughing: 
```
Setting status of a8deb61127ef3de8e5b48c034e2f3c007a23ac10 to FAILURE with url https://jenkins.hummingbird-tech.io/job/Testing/job/Python/job/hummingbirdtech-planting-lines/17/ and message: 'Jenkins Build FAILED
 No test results found.'
Using context: Jenkins
ERROR: /var/lib/jenkins/workspace/Testing/Python/hummingbirdtech-planting-lines@script/Jenkinsfile not found
Finished: FAILU
Setting status of a8deb61127ef3de8e5b48c034e2f3c007a23ac10 to FAILURE with url https://jenkins.hummingbird-tech.io/job/Testing/job/Python/job/hummingbirdtech-planting-lines/17/ and message: 'Jenkins Build FAILED
 No test results found.'
Using context: Jenkins
ERROR: /var/lib/jenkins/workspace/Testing/Python/hummingbirdtech-planting-lines@script/Jenkinsfile not found
Finished: FAILU
```
Please add a test at least to ensure the runnable is available, see one example:
https://github.com/HummingbirdTechGroup/hummingbirdtech-green-area-index/blob/develop/tests/green_area_index/runnable_test.py

**Comment by VirginieBfd** [comment]
@BlayneJLees I will review when the tests will pass, try out @LTMeyer's suggestion

**Comment by BlayneJLees** [comment]
> Jenkins tests are failing because they are considering the wrong branch. Have they succeeded when manually launched?

Ahh they were just working and then I committed again!

**Pull request merged**